src/libmongoc/tests/test-mongoc-sdam-monitoring.c

```diff
@@ -838,19 +841,17 @@ _test_heartbeat_fails_dns (bool pooled)
       client_set_heartbeat_event_callbacks (client, &context);
    }
 
-   start = bson_get_monotonic_time ();
-
    /* trigger "ismaster" handshake */
    r = mongoc_client_command_simple (
       client, "admin", tmp_bson ("{'foo': 1}"), NULL, NULL, &error);
 
+   /* This should result in either a DNS failure or connection failure depending
+    * on the network. */
    ASSERT (!r);
    ASSERT_ERROR_CONTAINS (error,
                           MONGOC_ERROR_SERVER_SELECTION,
                           MONGOC_ERROR_SERVER_SELECTION_FAILURE,
-                          "Failed to resolve");
-
-   duration = bson_get_monotonic_time () - start;
+                          "");
```

Comment:
Is this forgoing any assertion on the string message and just relying on a domain/code check? If so, I'd suggest amending the comment above to explain something like "We assert the domain/code but not the message string".